### PR TITLE
DRY out Phalanx

### PR DESCRIPTION
**Pull request by athornton**
(cachemachine first)

### CODE REVIEW

File: science-platform/templates/alert-stream-broker-application.yaml [hunk 1]
```diff
@@ -24,6 +24,14 @@ spec:
     repoURL: {{ .Values.repoURL }}
     targetRevision: {{ .Values.revision }}
     helm:
+      parameters:
+        - name: "global.host"
+          value: {{ .Values.fqdn | quote }}
+        - name: "global.baseUrl"
+          value: "https://{{ .Values.fqdn }}"
+        - name: "global.vaultSecretsPath"
```
Comment (jonathansick) [inline science-platform/templates/alert-stream-broker-application.yaml:32]: I think I called this `global.vaultSecretsPathPrefix` in my apps; I don't know if you want to do the same?

**Review by rra — CHANGES_REQUESTED** [review]
I'd rather we not add the global settings to every application without regard to whether they use them. Third-party applications mostly can't do anything useful with those settings, and will need other settings injected.

I'd rather we add the Helm parameters at the same time that we port a chart so that each change is self-contained.

File: science-platform/templates/alert-stream-broker-application.yaml (hunk 1 above)
Comment (rra) [reply science-platform/templates/alert-stream-broker-application.yaml:32]: I'd be good with that, but we should be consistent one way or the other across all applications, so we should probably fix the existing ones in one direction or the other.

File: services/datalinker/Chart.yaml [hunk 2]
```diff
@@ -1,10 +1,8 @@
...
-  - name: datalinker
-    version: 0.1.6
-    repository: https://lsst-sqre.github.io/charts/
-  - name: pull-secret
-    version: 0.1.2
-    repository: https://lsst-sqre.github.io/charts/
+type: application
+version: 0.1.6
+maintainers:
+  - name: cbanek
```
Comment (rra) [inline services/datalinker/Chart.yaml:8]: Delete maintainers when you move charts over.

File: services/datalinker/templates/NOTES.txt [hunk 3]
```diff
@@ -0,0 +1,22 @@
+1. Get the application URL by running these commands:
```
Comment (rra) [inline services/datalinker/templates/NOTES.txt:1]: Delete this file; it's only used for manual Helm installations and doesn't do anything useful in an Argo CD environment.

File: services/datalinker/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "datalinker.selectorLabels" . | nindent 8 }}
+    spec:
+      automountServiceAccountToken: false
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
```
Comment (rra) [inline services/datalinker/templates/deployment.yaml:27]: This should no longer be a template or in `values.yaml` and should instead be hard-coded with the value of `pull-secret` as created by the chart.

File: services/datalinker/templates/ingress.yaml [hunk 5]
```diff
@@ -0,0 +1,35 @@
+{{- if .Values.ingress.enabled -}}
```
Comment (rra) [inline services/datalinker/templates/ingress.yaml:1]: Remove this conditional and make the ingress unconditional.

File: services/datalinker/templates/ingress.yaml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  labels:
+    {{- include "datalinker.labels" . | nindent 4 }}
+  {{- with .Values.ingress.annotations }}
+  annotations:
+    kubernetes.io/ingress.class: "nginx"
+    {{- if .Values.ingress.gafaelfawrAuthQuery -}}
+    nginx.ingress.kubernetes.io/auth-method: GET
+    nginx.ingress.kubernetes.io/auth-response-headers: X-Auth-Request-User,X-Auth-Request-Email,X-Auth-Request-Token
+    nginx.ingress.kubernetes.io/auth-signin: "https://{{ .Values.global.host }}/login"
+    nginx.ingress.kubernetes.io/auth-url: "https://{{ .Values.global.host }}/auth?{{ .Values.ingress.gafaelfawrAuthQuery }}"
```
Comment (rra) [inline services/datalinker/templates/ingress.yaml:16]: Use `.Values.global.baseUrl` instead.

File: services/datalinker/templates/ingress.yaml [hunk 7]
```diff
@@ -0,0 +1,35 @@
+{{- if .Values.ingress.enabled -}}
+{{- $fullName := include "datalinker.fullname" . -}}
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: {{ $fullName }}
+  labels:
+    {{- include "datalinker.labels" . | nindent 4 }}
+  {{- with .Values.ingress.annotations }}
```
Comment (rra) [inline services/datalinker/templates/ingress.yaml:9]: Nit: Move this line down to where the annotations are actually used to make it match the pattern of other services.

File: services/datalinker/templates/ingress.yaml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    nginx.ingress.kubernetes.io/auth-response-headers: X-Auth-Request-User,X-Auth-Request-Email,X-Auth-Request-Token
+    nginx.ingress.kubernetes.io/auth-signin: "https://{{ .Values.global.host }}/login"
+    nginx.ingress.kubernetes.io/auth-url: "https://{{ .Values.global.host }}/auth?{{ .Values.ingress.gafaelfawrAuthQuery }}"
+    {{- end }}
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  {{- if .Values.ingress.className }}
+  ingressClassName: {{ .Values.ingress.className }}
+  {{- end }}
```
Comment (rra) [inline services/datalinker/templates/ingress.yaml:23]: This shouldn't be configurable any more.

File: services/datalinker/templates/ingress.yaml [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+  rules:
+    - host: {{ required "ingress.host must be set" .Values.global.host | quote }}
```
Comment (rra) [inline services/datalinker/templates/ingress.yaml:25]: Error message is now wrong (should be `global.host`).

File: services/datalinker/templates/ingress.yaml [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+      http:
+        paths:
+          - path: {{ .Values.ingress.path }}
+            pathType: {{ default "Prefix" .Values.ingress.pathType }}
```
Comment (rra) [inline services/datalinker/templates/ingress.yaml:29]: These shouldn't be configurable.

File: services/datalinker/templates/ingress.yaml [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+            backend:
+              service:
+                name: {{ $fullName }}
+                port:
+                  number: {{ .Values.service.port }}
```
Comment (rra) [inline services/datalinker/templates/ingress.yaml:34]: This shouldn't be configurable; just hard-code the correct value.

File: services/datalinker/templates/service.yaml [hunk 12]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "datalinker.fullname" . }}
+  labels:
+    {{- include "datalinker.labels" . | nindent 4 }}
+spec:
+  type: {{ .Values.service.type }}
+  ports:
+    - port: {{ .Values.service.port }}
```
Comment (rra) [inline services/datalinker/templates/service.yaml:10]: Hard-code these values now.

File: services/datalinker/templates/vault-pull-secret.yaml [hunk 13]
```diff
@@ -0,0 +1,9 @@
+apiVersion: ricoberger.de/v1alpha1
```
Comment (rra) [inline services/datalinker/templates/vault-pull-secret.yaml:1]: This file should be named `vault-secrets.yaml` to match the other applications.

File: services/datalinker/values.yaml [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+
+  # -- Pull policy for the datalinker image
+  pullPolicy: Always
+
+  # -- Overrides the image tag whose default is the chart appVersion.
+  tag: ""
+
+# -- Secret names to use for all Docker pulls
+imagePullSecrets: 
+  - name: "pull-secret"
```
Comment (rra) [inline services/datalinker/values.yaml:20]: Delete.

File: services/datalinker/Chart.yaml (hunk 2 above)
Comment (athornton) [reply services/datalinker/Chart.yaml:8]: OK.

File: services/datalinker/values-idfdev.yaml [hunk 15]
```diff
@@ -1,8 +1 @@
-datalinker:
-  ingress:
-    enabled: true
-    host: "data-dev.lsst.cloud"
-
-pull-secret:
-  enabled: true
-  path: secret/k8s_operator/data-dev.lsst.cloud/pull-secret
+# idfdev
```
[the remark this answers is not in the document]
Comment (athornton) [reply services/datalinker/values-idfdev.yaml:1]: `git push` wasn't pushing zero-length files for me.  Is that a git setting I have somewhere?

**Comment by athornton** [comment]
> I'd rather we not add the global settings to every application without regard to whether they use them. Third-party applications mostly can't do anything useful with those settings, and will need other settings injected.
> 
> I'd rather we add the Helm parameters at the same time that we port a chart so that each change is self-contained.

OK, backing out the ones I haven't done yet.

File: science-platform/templates/alert-stream-broker-application.yaml (hunk 1 above)
Comment (athornton) [reply science-platform/templates/alert-stream-broker-application.yaml]: Y'all figure out which way it should go, and I'm happy to make it consistent.  I prefer the shorter version, myself.

File: services/datalinker/templates/ingress.yaml (hunk 11 above)
Comment (athornton) [reply services/datalinker/templates/ingress.yaml]: done

File: services/datalinker/templates/service.yaml (hunk 12 above)
Comment (athornton) [reply services/datalinker/templates/service.yaml]: done

File: services/datalinker/templates/NOTES.txt (hunk 3 above)
Comment (athornton) [reply services/datalinker/templates/NOTES.txt]: done

File: services/datalinker/values-idfdev.yaml (hunk 15 above)
[the remark this answers is not in the document]
Comment (athornton) [reply services/datalinker/values-idfdev.yaml]: Musta been PEBCAK

File: services/datalinker/values.yaml [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+
+# -- Override the base name for resources
+nameOverride: ""
+
+# -- Override the full name for resources (includes the release name)
+fullnameOverride: ""
```
[the remark this answers is not in the document]
Comment (athornton) [reply services/datalinker/values.yaml]: done

File: services/datalinker/values.yaml [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+  gafaelfawrAuthQuery: ""
+
+  # -- Additional annotations for the ingress rule
+  annotations: {}
+
+  # -- Path type for the ingress rule
+  pathType: ImplementationSpecific
+
+  # -- URL path to dispatch to the datalinker deployment pod
+  path: "/api/datalink"
```
[the remark this answers is not in the document]
Comment (athornton) [reply services/datalinker/values.yaml]: done

File: services/datalinker/values.yaml (hunk 14 above)
Comment (athornton) [reply services/datalinker/values.yaml]: done

File: services/datalinker/templates/vault-pull-secret.yaml (hunk 13 above)
Comment (athornton) [reply services/datalinker/templates/vault-pull-secret.yaml]: done

File: services/datalinker/templates/ingress.yaml (hunk 9 above)
Comment (athornton) [reply services/datalinker/templates/ingress.yaml]: done

File: services/datalinker/templates/ingress.yaml (hunk 7 above)
Comment (athornton) [reply services/datalinker/templates/ingress.yaml]: done

**Comment by athornton** [comment]
I think it's ready to look at again.  I believe I did all the stuff Russ asked, consistently (I hope) across repositories.

The sasquatch/strimzi-registry-operator interaction is still kind of opaque to me, and nublado2 would need some code changes if we wanted to REALLY get rid of the pull-and-butler-secret config in the environment values files.

I also removed pull secrets from those repositories (mobu, argocd, semaphore, squareone) where we do not need them and we are not pulling from Docker Hub or GAR or T&S Nexus.

**Comment by athornton** [comment]
It may be easier to look at this commit-by-commit, which largely tracks individual applications' migration.

**Review by afausti — CHANGES_REQUESTED** [review]
Note that `strimzi-registry-operator` is an example of a service that is shared among other phalanx services, and per RFC-830 it should be located in phalanx/charts/ and symlinked from the services that use it. I think this should be done in a separate PR along with the changes in sasquatch and the alert-stream-broker.

**Comment by athornton** [comment]
I think I did what Angelo requested; I am not currently testing in a place with strimzi-registry-operator enabled, though.

File: services/exposurelog/values.yaml [hunk 18]
```diff
@@ -1,6 +1,8 @@
 # Default values for exposurelog.
```
Comment (rra) [inline services/exposurelog/values.yaml:1]: Needs helm-docs comments added for all the settings.

File: services/moneypenny/README.md [hunk 19]
```diff
@@ -0,0 +1,38 @@
+# moneypenny
```
Comment (rra) [inline services/moneypenny/README.md:1]: Needs `README.md.gotmpl`

File: services/moneypenny/templates/ingress.yaml [hunk 20]
```diff
@@ -0,0 +1,36 @@
...
+  {{- if .Values.ingress.tls }}
+  tls:
+    {{- range .Values.ingress.tls }}
+    - hosts:
+      {{- range .hosts }}
+        - {{ . | quote }}
+      {{- end }}
+      secretName: {{ .secretName }}
+    {{- end }}
+  {{- end }}
```
Comment (rra) [inline services/moneypenny/templates/ingress.yaml:36]: Delete all the TLS stuff.

File: services/moneypenny/templates/serviceaccount.yaml [hunk 21]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ include "moneypenny.serviceAccountName" . }}
+  labels:
+    {{- include "moneypenny.labels" . | nindent 4 }}
+imagePullSecrets:
+  - name: "pull-secret"
```
Comment (rra) [inline services/moneypenny/templates/serviceaccount.yaml:8]: Similarly pointless here. This is already set on the deployments.

File: services/narrativelog/README.md [hunk 22]
```diff
@@ -1,6 +1,38 @@
 # narrativelog
```
Comment (rra) [inline services/narrativelog/README.md:1]: Needs `README.md.gotmpl`

File: services/narrativelog/values.yaml [hunk 23]
```diff
@@ -1,6 +1,8 @@
 # Default values for narrativelog.
```
Comment (rra) [inline services/narrativelog/values.yaml:1]: Needs helm-docs comments for all the settings.

File: services/obstap/README.md [hunk 24]
```diff
@@ -0,0 +1,47 @@
+# cadc-tap-postgres
```
Comment (rra) [inline services/obstap/README.md:1]: Needs `README.md.gotmpl`

File: services/plot-navigator/README.md [hunk 25]
```diff
@@ -0,0 +1,18 @@
+# plot-navigator
```
Comment (rra) [inline services/plot-navigator/README.md:1]: Needs `README.md.gotmpl`

File: services/plot-navigator/templates/deployment.yaml [hunk 26]
```diff
@@ -0,0 +1,66 @@
+---
```
Comment (rra) [inline services/plot-navigator/templates/deployment.yaml:1]: Stray document separator.

File: services/plot-navigator/templates/deployment.yaml [hunk 27]
```diff
@@ -0,0 +1,66 @@
...
+    {{- include "plot-navigator.labels" . | nindent 4 }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: plot-navigator
+  template:
+    metadata:
+      labels:
+        app: plot-navigator
```
Comment (rra) [inline services/plot-navigator/templates/deployment.yaml:16]: Should use the standard selector labels.

File: services/plot-navigator/templates/ingress.yaml [hunk 28]
```diff
@@ -0,0 +1,30 @@
+---
```
Comment (rra) [inline services/plot-navigator/templates/ingress.yaml:1]: Stray document separator

File: services/plot-navigator/templates/service.yaml [hunk 29]
```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: plot-navigator
+  labels:
+    {{- include "plot-navigator.labels" . | nindent 4 }}
+spec:
+  selector:
+    app: plot-navigator
```
Comment (rra) [inline services/plot-navigator/templates/service.yaml:9]: Should use the standard selector labels

File: services/plot-navigator/templates/vault-secrets.yaml [hunk 30]
```diff
@@ -0,0 +1,20 @@
+---
```
Comment (rra) [inline services/plot-navigator/templates/vault-secrets.yaml:1]: Stray document separator

File: services/postgres/README.md [hunk 31]
```diff
@@ -0,0 +1,21 @@
+# postgres
```
Comment (rra) [inline services/postgres/README.md:1]: Needs `README.md.gotmpl`

File: services/postgres/templates/_helpers.tpl [hunk 32]
```diff
@@ -0,0 +1,56 @@
...
+{{/*
+Create the name of the service account to use
+*/}}
+{{- define "postgres.serviceAccountName" -}}
+{{- if .Values.serviceAccount.create -}}
+    {{ default (include "postgres.fullname" .) .Values.serviceAccount.name }}
+{{- else -}}
+    {{ default "default" .Values.serviceAccount.name }}
+{{- end -}}
+{{- end -}}
```
Comment (rra) [inline services/postgres/templates/_helpers.tpl:56]: Missing the standard selectorLabels template

File: services/postgres/templates/deployment.yaml [hunk 33]
```diff
@@ -0,0 +1,95 @@
...
+    app: {{ template "postgres.fullname" . }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      name: {{ template "postgres.fullname" . }}
+  template:
+    metadata:
+      labels:
+        name: {{ template "postgres.fullname" . }}
```
Comment (rra) [inline services/postgres/templates/deployment.yaml:15]: Should use the standard selector labels

File: services/postgres/templates/deployment.yaml [hunk 34]
```diff
@@ -0,0 +1,95 @@
...
+    spec:
+      containers:
+        - name: {{ template "postgres.fullname" . }}
+          imagePullPolicy: "Always"
+          image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
```
Comment (rra) [inline services/postgres/templates/deployment.yaml:20]: Should default to the appVersion

File: services/postgres/templates/physpvc.yaml [hunk 35]
```diff
@@ -0,0 +1,16 @@
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
```
Comment (rra) [inline services/postgres/templates/physpvc.yaml:3]: Missing standard labels

File: services/postgres/templates/physpvc.yaml [hunk 36]
```diff
@@ -0,0 +1,16 @@
...
+    - "ReadWriteOnce"
+  resources:
+    requests:
+      storage: {{.Values.postgres_volume_size}}
+  storageClassName: {{.Values.postgres_storage_class}}
+{{ if .Values.volume_name }}
+  volumeName: {{ .Values.volume_name }}
+{{ end }}
+
+
```
Comment (rra) [inline services/postgres/templates/physpvc.yaml:16]: Stray blank lines and should use `-` to delete unwanted whitespace for the `if` and `end` macros.

File: services/postgres/templates/service.yaml [hunk 37]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: Service
+metadata:
```
Comment (rra) [inline services/postgres/templates/service.yaml:3]: Missing standard labels

File: services/postgres/templates/service.yaml [hunk 38]
```diff
@@ -0,0 +1,10 @@
...
+  name: {{ template "postgres.fullname" . }}
+spec:
+  ports:
+    - name: postgres
+      port: 5432
+  selector:
+    name: {{ template "postgres.fullname" . }}
```
Comment (rra) [inline services/postgres/templates/service.yaml:10]: Should use standard selector labels

File: services/postgres/templates/storageclass.yaml [hunk 39]
```diff
@@ -0,0 +1,9 @@
+{{- if eq .Values.postgres_storage_class "fast" }}
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: fast
+provisioner: kubernetes.io/gce-pd
+parameters:
+  type: pd-ssd
+{{- end }}
```
Comment (rra) [inline services/postgres/templates/storageclass.yaml:9]: Do we care about this any more? If we want a good backend, wouldn't we just use Cloud SQL?

File: services/postgres/values-base.yaml [hunk 40]
```diff
@@ -1,25 +1,16 @@
...
-  narrativelog_db:
-    user: 'narrativelog'
-    db: 'narrativelog'
-  postgres_storage_class: 'rook-ceph-block'
-
-pull-secret:
-  enabled: true
-  path: secret/k8s_operator/base-lsp.lsst.codes/pull-secret
+jupyterhub_db:
+  user: 'jovyan'
```
Comment (rra) [inline services/postgres/values-base.yaml:2]: Here and in the other values files, these are using single quotes, but we use double quotes everywhere else. While we're doing this sort of big refactoring, we should probably be consistent.

File: services/postgres/values.yaml [hunk 41]
```diff
@@ -0,0 +1,18 @@
+# Default values for fileserver.
```
Comment (rra) [inline services/postgres/values.yaml:1]: Needs helm-docs comments. The standard for Helm parameters is studlyCaps, not snake_case. Looks like most of the parameters aren't documented and probably should be.

File: services/sherlock/README.md [hunk 42]
```diff
@@ -0,0 +1,33 @@
+# sherlock
```
Comment (rra) [inline services/sherlock/README.md:1]: Needs `README.md.gotmpl`

File: services/sherlock/values.yaml [hunk 43]
```diff
@@ -0,0 +1,67 @@
...
+tolerations: []
+
+# -- Affinity rules for the sherlock deployment pod
+affinity: {}
+
+serviceAccount:
+  name: ""
+
+# -- URL to push status to via HTTP PUTs.
+publish_url: ""
```
Comment (rra) [inline services/sherlock/values.yaml:67]: Should use studlyCaps here as well.

File: services/strimzi-registry-operator/Chart.yaml [hunk 44]
```diff
@@ -1,7 +1,5 @@
 apiVersion: v2
```
Comment (rra) [inline services/strimzi-registry-operator/Chart.yaml:1]: I'm confused; strimzi-registry-operator was added as a shared chart earlier. It shouldn't be both a service and a shared chart; it should be one or the other, no?

File: services/strimzi-registry-operator/values.yaml [hunk 45]
```diff
@@ -0,0 +1,11 @@
+image:
```
Comment (rra) [inline services/strimzi-registry-operator/values.yaml:1]: Missing helm-docs comments for several parameters.

File: services/tap/README.md [hunk 46]
```diff
@@ -0,0 +1,54 @@
+# cadc-tap
```
Comment (rra) [inline services/tap/README.md:1]: Needs `README.md.gotmpl`

**Review by rra — COMMENTED** [review]
I didn't check all the `values.yaml` files, but I think some of the ones you moved over from charts are missing the `values.yaml` entries for the global parameters injected by Argo CD. This will cause test failures from the Helm linter.

**Comment by athornton** [comment]
Did some more linting.  I don't understand why the linter is failing on strimzi-request-operator but it looks like it doesn't know it's supposed to template it?  Maybe because it's not in a "templates" directory outside the symlink?  I'll ask Angelo tomorrow.

**Comment by athornton** [comment]
Well, argocd is gonna need some work.  Trying to DRY it and make it use global secret path is hanging in app installation.  So...let's punt that till later.  This work does quite a bit of cleanup for our other services, anyway.

**Comment by rra** [comment]
The strimzi-registry-operator chart isn't valid if I'm reading this correctly. It needs the normal stuff like a `Chart.yaml` and a `values.yaml`.

I'm confused by why there's still a separate `services/strimzi-registry-operator` chart. The users of the strimzi-registry-operator chart should have a symlink in their `charts` subdirectory that points to `charts/strimzi-registry-operator`.

**Comment by athornton** [comment]
Now *I* am confused.  I haven't deployed strimzi-registry-operator, apparently.  I thought the minikube deployment was doing so since it has values-minikube.yaml but it's got enabled=false in science-platform.

However, when I run
```
adam@m1-wired:~/git/phalanx/services/strimzi-registry-operator$ helm template -f values.yaml -f values-minikube.yaml .
```
I get a warning, but that warning is accurate, but it produces what looks like a reasonable manifest.
```
walk.go:74: found symbolic link in path: /Users/adam/git/phalanx/services/strimzi-registry-operator/templates resolves to /Users/adam/git/phalanx/charts/strimzi-registry-operator/templates
```

**Comment by athornton** [comment]
OK....strimzi-registry-operator is back to master.  I could try changing the repository to point to the local charts directory (for that and sasquatch) if we want, or we could this good enough.

File: services/datalinker/templates/ingress.yaml [hunk 47]
```diff
@@ -0,0 +1,30 @@
...
+  name: {{ $fullName }}
+  labels:
+    {{- include "datalinker.labels" . | nindent 4 }}
+spec:
+  rules:
+    - host: {{ required "global.host must be set" .Values.global.host | quote }}
+      http:
+        paths:
+          - path: "/api/datalink"
+            pathType: ImplementationSpecific
```
Comment (rra) [inline services/datalinker/templates/ingress.yaml:25]: `pathType` should be `Prefix`

File: services/obstap/Chart.yaml [hunk 48]
```diff
@@ -1,10 +1,6 @@
...
-name: obstap
-version: 1.0.0
-dependencies:
-- name: cadc-tap-postgres
-  version: 0.2.2
-  repository: https://lsst-sqre.github.io/charts/
-- name: pull-secret
-  version: 0.1.2
-  repository: https://lsst-sqre.github.io/charts/
+apiVersion: v1
```
Comment (rra) [inline services/obstap/Chart.yaml:1]: `apiVersion` should be `v2`

File: services/obstap/templates/tap-ingress-anonymous.yaml [hunk 49]
```diff
@@ -0,0 +1,45 @@
...
+                name: {{ template "cadc-tap-postgres.fullname" . }}
+                port:
+                  number: 80
+          - path: "/api/obstap/(swagger-ui.*)"
+            pathType: "ImplementationSpecific"
+            backend:
+              service:
+                name: {{ template "cadc-tap-postgres.fullname" . }}
+                port:
+                  number: 80
```
Comment (rra) [inline services/obstap/templates/tap-ingress-anonymous.yaml:45]: This is already a regex, so these can all be merged into one rule, such as for the tap service.

File: services/plot-navigator/values.yaml [hunk 50]
```diff
@@ -0,0 +1,26 @@
...
+  tag: ""
+
+# Environment variables to be passed to panel.
+# e.g. butler configuration and auth params.
+environment: {}
+
+ingress:
+  gafaelfawrAuthQuery: "scope=exec:portal&delegate_to=plotnavigator"
+  annotations: {}
+
```
Comment (rra) [inline services/plot-navigator/values.yaml:12]: Need helm-docs comments

**Review by rra — COMMENTED** [review]
There are still files in `charts/strimzi-registry-operator`. Presumably those should now be deleted?

I've resolved the inline comments that were obsolete. The remaining ones from the previous review all still need to be addressed.

**Comment by athornton** [comment]
Gaaah.  `git status` might have been helpful to me.

File: services/datalinker/templates/ingress.yaml (hunk 47 above)
Comment (athornton) [reply services/datalinker/templates/ingress.yaml]: Done.

File: services/obstap/Chart.yaml (hunk 48 above)
Comment (athornton) [reply services/obstap/Chart.yaml]: Done.

File: services/obstap/templates/tap-ingress-anonymous.yaml (hunk 49 above)
Comment (athornton) [reply services/obstap/templates/tap-ingress-anonymous.yaml]: Done.

File: services/plot-navigator/values.yaml (hunk 50 above)
Comment (athornton) [reply services/plot-navigator/values.yaml]: Helm docs added.  Charts deleted.

File: science-platform/templates/postgres-application.yaml [hunk 51]
```diff
@@ -21,6 +21,14 @@ spec:
     repoURL: {{ .Values.repoURL }}
     targetRevision: {{ .Values.revision }}
     helm:
+      parameters:
+        - name: "global.host"
+          value: {{ .Values.fqdn | quote }}
+        - name: "global.baseUrl"
+          value: "https://{{ .Values.fqdn }}"
```
[the remark this answers is not in the document]
Comment (athornton) [reply science-platform/templates/postgres-application.yaml:28]: Done.

File: services/cachemachine/templates/serviceaccount.yaml [hunk 52]
```diff
@@ -8,6 +8,8 @@ metadata:
   annotations:
     {{- toYaml . | nindent 4 }}
   {{- end }}
+imagePullSecrets:
+  - name: "pull-secret"
```
[the remark this answers is not in the document]
Comment (athornton) [reply services/cachemachine/templates/serviceaccount.yaml:12]: Done

File: services/exposurelog/values.yaml (hunk 18 above)
Comment (athornton) [reply services/exposurelog/values.yaml:1]: Done

File: services/moneypenny/templates/ingress.yaml (hunk 20 above)
Comment (athornton) [reply services/moneypenny/templates/ingress.yaml:36]: Done

File: services/moneypenny/templates/serviceaccount.yaml (hunk 21 above)
Comment (athornton) [reply services/moneypenny/templates/serviceaccount.yaml:8]: imagePullSecrets removed

File: services/narrativelog/values.yaml (hunk 23 above)
Comment (athornton) [reply services/narrativelog/values.yaml:1]: Done.

File: services/plot-navigator/templates/deployment.yaml (hunk 26 above)
Comment (athornton) [reply services/plot-navigator/templates/deployment.yaml:1]: done

File: services/plot-navigator/templates/deployment.yaml (hunk 27 above)
Comment (athornton) [reply services/plot-navigator/templates/deployment.yaml:16]: Done

File: services/postgres/templates/_helpers.tpl (hunk 32 above)
Comment (athornton) [reply services/postgres/templates/_helpers.tpl:56]: Done

File: services/postgres/templates/deployment.yaml (hunk 33 above)
Comment (athornton) [reply services/postgres/templates/deployment.yaml:15]: Done

File: services/postgres/templates/deployment.yaml (hunk 34 above)
Comment (athornton) [reply services/postgres/templates/deployment.yaml:20]: Done

File: services/postgres/templates/physpvc.yaml (hunk 35 above)
Comment (athornton) [reply services/postgres/templates/physpvc.yaml:3]: Done

File: services/postgres/templates/physpvc.yaml (hunk 36 above)
Comment (athornton) [reply services/postgres/templates/physpvc.yaml:16]: Done

File: services/postgres/templates/storageclass.yaml (hunk 39 above)
Comment (athornton) [reply services/postgres/templates/storageclass.yaml:9]: That's probably the right thing, but our current postgres implementations are all using "fast" and putting the postgres volume on SSD currently.

File: services/postgres/values-base.yaml (hunk 40 above)
Comment (athornton) [reply services/postgres/values-base.yaml:2]: Done

File: services/postgres/values.yaml (hunk 41 above)
Comment (athornton) [reply services/postgres/values.yaml:1]: Done (and converted to studlyCaps)

File: services/sherlock/values.yaml (hunk 43 above)
Comment (athornton) [reply services/sherlock/values.yaml:67]: Done.

**Comment by athornton** [comment]
I think this time, maybe, I actually HAVE addressed all the review comments.  At this point the only thing I know of that is still outstanding is:

global.vaultSecretsPathPrefix or global.vaultSecretsPath ?

I prefer the shorter version, but I don't really care.  @rra and @jonathansick, please pick a winner between the two of those, and I'll go make it consistent.

**Review by rra — APPROVED** [review]

**Review by afausti — APPROVED** [review]

**Pull request merged**